### PR TITLE
changes in the attachments is fixed

### PR DESCRIPTION
**Pull request by farhanp1502**

### CODE REVIEW

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts [hunk 1]
```diff
@@ -42,20 +42,17 @@ export class DetailsPageComponent implements OnInit {
...
       this.submitted = data.data.status == statusType.submitted
       this.getProjectTaskStatus()
-      this.countCompletedTasks(this.projectDetails);
+      this.countCompletedTasks();
       this.calculateProgress();
       this.setActionsList()
       this.initializeTasks()
     })
   }
+  countCompletedTasks(): number {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:51]: Why is this function added?

File: projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.ts [hunk 2]
```diff
@@ -36,8 +36,8 @@ export class AttachmentListingPageComponent {
     let result = { project: false, task: false };
     if(attachmentType === 'image'){
       if ( this.attachments.project || this.attachments.tasks ) {
-          result.project = this.attachments.project.attachments && this.attachments.project.attachments.some((attachment: any) => attachment.type.includes(attachmentType)) || this.attachments.project.remarks;
-          result.task = this.attachments.tasks.some((task: any) => task.attachments && task.attachments.some((attachment: any) => attachment.type.includes(attachmentType))) || this.attachments.tasks.some((task:any)=>task.remarks);
+        result.project = !!(this.attachments.project.attachments && this.attachments.project.attachments.some((attachment: any) => attachment.type.includes(attachmentType)) || this.attachments.project.remarks);
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.ts:39]: What is this condition ? (!!)

File: projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html [hunk 3]
```diff
@@ -1,8 +1,10 @@
 @if (attachments.type.includes('image')) {
     <mat-card class="w-full  rounded-full p-2 " style=" box-shadow: rgba(133, 119, 119, 0.555) 5px 5px 4px;">
-        <div class="flex w-full justify-end">
-            <mat-icon  class="primary-icon cursor-pointer" fontIcon="cancel" (click)="actionEmit(attachments)"></mat-icon>
-        </div>
+        @if ( projectDataStatus !== 'submitted' ) {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html:3]: Use the 'submitted' value from the constant file

File: projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html [hunk 4]
```diff
@@ -25,9 +27,11 @@
                 {{attachments.name}}
             </div>
         </div>
-        <div class="flex flex-col cursor-pointer" (click)="actionEmit(attachments)">
-            <div class="flex justify-center items-end"><mat-icon fontIcon="cancel" class="primary-icon"></mat-icon></div>
-            <div class="text-xs primary-icon">{{ "REMOVE" | translate}}</div>
-        </div>
+        @if ( projectDataStatus !== 'submitted' ) {
```
Comment (Vinod-V3) [inline projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html:30]: Use the 'submitted' value from the constant file

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts (hunk 1 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:51]: to should the number of task completed

File: projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.ts (hunk 2 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/attachment-listing-page/attachment-listing-page.component.ts:39]: !! converts this final truthy or falsy value to true or false.

File: projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html (hunk 3 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html]: done

File: projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html (hunk 4 above)
Comment (farhanp1502) [reply projects/project-player-library/src/lib/shared/attachment-show-card/attachment-show-card.component.html]: done

File: projects/project-player-library/src/lib/pages/details-page/details-page.component.ts (hunk 1 above)
Comment (Vinod-V3) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:51]: Arent we doing that in calculatePregress() ?

Comment (farhanp1502) [reply projects/project-player-library/src/lib/pages/details-page/details-page.component.ts:51]: no its just to calculate the progress value that's it

**Pull request merged**